web/src/user_profile.js

```diff
@@ -29,6 +30,8 @@ import * as user_pill from "./user_pill";
 import * as util from "./util";
 
 let user_streams_list_widget;
+let user_profile_subscribe_widget;
+let user_id;
```

Comment:
This variable feels unnecessary, and is also badly named -- we have several local variables by the same name in the file.

It would not be crazy to have a `currently_active_user_id` variable instead of parsing it from the DOM every time, but I decided to go with the other approach since that's what we're already doing in other code paths. Here's the diff:

``` diff
diff --git a/web/src/user_profile.js b/web/src/user_profile.js
index 5162684675..b58842bc91 100644
--- a/web/src/user_profile.js
+++ b/web/src/user_profile.js
@@ -31,7 +31,6 @@ import * as util from "./util";
 
 let user_streams_list_widget;
 let user_profile_subscribe_widget;
-let user_id;
 
 function compare_by_name(a, b) {
     return util.strcmp(a.name, b.name);
@@ -99,8 +98,9 @@ function change_state_of_subscribe_button(event, dropdown) {
 }
 
 export function get_user_unsub_streams() {
+    const target_user_id = Number.parseInt($("#user-profile-modal").attr("data-user-id"), 10);
     return stream_data
-        .get_streams_for_user(user_id)
+        .get_streams_for_user(target_user_id)
         .can_subscribe.filter((stream) => stream_data.can_subscribe_others(stream))
         .map((stream) => ({
             name: stream.name,
@@ -250,7 +250,6 @@ function initialize_user_type_fields(user) {
 
 export function show_user_profile(user, default_tab_key = "profile-tab") {
     popovers.hide_all();
-    user_id = user.user_id;
 
     const field_types = page_params.custom_profile_field_types;
     const profile_data = page_params.custom_profile_fields
@@ -372,6 +371,7 @@ export function register_click_handlers() {
         e.preventDefault();
         const stream_id = Number.parseInt(user_profile_subscribe_widget.value(), 10);
         const sub = sub_store.get(stream_id);
+        const target_user_id = Number.parseInt($("#user-profile-modal").attr("data-user-id"), 10);
         const $alert_box = $("#user-profile-streams-tab .stream_list_info");
         function addition_success(data) {
             if (Object.keys(data.subscribed).length > 0) {
@@ -391,7 +391,12 @@ export function register_click_handlers() {
         function addition_failure(xhr) {
             ui_report.error("", xhr, $alert_box, 1200);
         }
-        subscriber_api.add_user_ids_to_stream([user_id], sub, addition_success, addition_failure);
+        subscriber_api.add_user_ids_to_stream(
+            [target_user_id],
+            sub,
+            addition_success,
+            addition_failure,
+        );
     });
 
     $("body").on("click", "#user-profile-modal .remove-subscription-button", (e) => {
```